File: servidor-express/src/models/userModel.js
const database = require('../database')

exports.ListarUsuarios = async () => {
    const sql = "SELECT u.id_usuario ,u.nombre, u.apellido, u.correo, u.password, u.id_rol, r.descripcion FROM usuarios u, roles r WHERE u.id_rol = r.id_rol"
    return await database.query(sql)
}

exports.EliminarUsuario = async (params) => {
    const { id } = params
    const sql = "DELETE FROM usuarios WHERE id_usuario = ?"
    return await database.query(sql, [id])
}

exports.ActualizarUsuario = async (params, body) => {
    const { id } = params
    const { nombre, apellido, correo, password, id_rol } = body
    const sql = "UPDATE usuarios SET nombre = ?, apellido = ?, correo = ?, password = ?, id_rol = ? WHERE id_usuario = ?"
    return await database.query(sql, [nombre, apellido, correo, password, id_rol, id])
}

exports.AgregarUsuario = async (body) => {
    const { nombre, apellido, correo, password, id_rol } = body
    const sql = "INSERT INTO usuarios(nombre,apellido,correo,password,id_rol) VALUES (?,?,?,?,?)"
    return await database.query(sql, [nombre, apellido, correo, password, id_rol])
}

exports.LoginUsuario = async (body) => {
    const { correo, password } = body
    const sql = "SELECT * FROM usuarios WHERE correo = ? AND password = ?"
    return await database.query(sql, [correo, password])
}

exports.ObtenerDatosUsuario = async (params) => {
    const { correo, password } = params
    const sql = "SELECT id_usuario,nombre,apellido,correo,password,id_rol FROM usuarios WHERE correo = ? AND password = ?"
    return await database.query(sql, [correo, password])
}

File: movil/lib/models/user_model.dart
class User {
  int idUsuario;
  String nombre;
  String apellido;
  String correo;
  String password;
  int idRol;
  String descripcion;

  User({
      this.idUsuario,
      this.nombre,
      this.apellido,
      this.correo,
      this.password,
      this.idRol,
      this.descripcion
  });

  User.fromJson(Map<String, dynamic> json) {
    idUsuario = json['id_usuario'];
    nombre = json['nombre'];
    apellido = json['apellido'];
    correo = json['correo'];
    password = json['password'];
    idRol = json['id_rol'];
    descripcion = json['descripcion'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id_usuario'] = this.idUsuario;
    data['nombre'] = this.nombre;
    data['apellido'] = this.apellido;
    data['correo'] = this.correo;
    data['password'] = this.password;
    data['id_rol'] = this.idRol;
    data['descripcion'] = this.descripcion;
    return data;
  }
}
